acceptancetests/deploy_stack.py

```diff
@@ -570,6 +571,46 @@ def tear_down(self, has_controller):
             self.tear_down_client.kill_controller(check=True)
 
 
+class ExistingController:
```

Comment:
Unit tests for this would be useful, as it's actually part of the framework, not a testcase itself.